Toggle  categories and subcategories externally

kahummer opened the pull request:
Fix: https://github.com/onaio/irfsomalia/issues/896
This pr is connected depends on
https://github.com/onaio/gisida/pull/485, 
https://github.com/onaio/gisida-react-private/pull/80
![Screenshot from 2020-08-05 23-58-47](https://user-images.githubusercontent.com/16018123/89463456-a2c32700-d777-11ea-9305-7696789e23a0.png)



- [x] Add ability to close annexed legends on menu/legend

- [x] Refactor menu to close categories and subcategories externally (from gisida-react-private)

- [x] Fix tests
https://www.awesomescreenshot.com/video/683573?key=f4154f608237741522a5bd49a1a33773

kahummer committed:
Close layer on legend

kahummer committed:
Toggle sub-menus externally

kelvin-muchiri commented on src/lib/utils.js:
```diff
@@ -369,3 +369,90 @@ export function getSharedStyleFromURL(mapId) {
 
   return style ? +style : null;
 }
+
```
Can we add docstrings here and tests for this util method

kelvin-muchiri commented on src/lib/utils.js:
```diff
@@ -267,26 +267,26 @@ export function getSharedLayersFromURL(mapId) {
-    return children.map(child => child.id);
-  }
-}
+// export function getMenuGroupMapLayers(groupName, children) {
```
Why was this commented? I can see the functionality has been instead copied into the function `getCategoryForLayers`

kelvin-muchiri commented on src/lib/utils.js:
```diff
@@ -369,3 +369,90 @@ export function getSharedStyleFromURL(mapId) {
 }
+
+export function getCategoryForLayers(layersToOpenCategory, categories) {
+  const  getMenuGroupMapLayers = (groupName, children) => {
```
Can we add docstrings here and tests for this util method

kelvin-muchiri replied:
`getMenuGroupMapLayers` This should be moved back to its util method

kelvin-muchiri commented on src/lib/utils.js:
```diff
@@ -369,3 +369,90 @@ export function getSharedStyleFromURL(mapId) {
   return style ? +style : null;
 }
+
+export function getCategoryForLayers(layersToOpenCategory, categories) {
```
Can we add docstrings here and tests for this util method

kahummer committed:
Render legends based on activeLayerIds

kelvin-muchiri commented on src/lib/utils.js:
```diff
@@ -369,3 +369,90 @@ export function getSharedStyleFromURL(mapId) {
+  const layerCategory = [];
+  if (layersToOpenCategory && categories) {
+    layersToOpenCategory
+      .filter(l => !l.isCatOpen)
```
`layersToOpenCategory` should be an array of layer id strings not objects. `.filter(l => !l.isCatOpen)` is not required here since the purpose of this util to know what category each layer id in the array falls under

kelvin-muchiri committed:
Get menu categories from state

As part of performance enhancement, the following changes were made

1. Menu categories are now read from store instead of being built inside the Menu component
2. We wait for map-1 to load before loading map-2 as this was also contributing to the slow performance

Changes in components Layer, Layers, ConnectedLayers underwent changes because unnecessary properties in the category layer objects were removed. The only properties required for the menu categories are id, label and category

kelvin-muchiri committed:
Solve bug layer not loading

kelvin-muchiri committed:
Rename variable

kelvin-muchiri committed:
Remove implementation to add id for menu groups

The addition of a group id is now done during building of the menu categories in gisida

kelvin-muchiri committed:
Solve bug layer checked by default

kelvin-muchiri committed:
Refactor code

kahummer committed:
Update legend style

kelvin-muchiri committed:
Make use of menu group parent instead of count for ConnectedLayers

The group count was used to make sure group names with the same name under different categories do not open/close if one is opened/closed. The same solution can be achieved by using the newly added parent attribute for each group

kelvin-muchiri committed:
Remove console.log, add build to gitignore

kelvin-muchiri committed:
Refactor Menu and add tests for components/Menu/utils/canAccessLayer

kahummer committed:
Menu connected layers tweaks

kelvin-muchiri committed:
Add tests for Menu uitl methods getAccessibleGroupLayer, getAccessibleCategories

kelvin-muchiri committed:
Update doc strings and add test case for Menu util method getAccessibleCategories

kahummer committed:
Merge branch 'toggle-sectors-externally' of github.com:onaio/gisida-react into toggle-sectors-externally

kahummer committed:
Category & menu groups toggle fixes

kelvin-muchiri committed:
Get Menu categories from state

categories received from props required to filtered in order to return only the ones the user has access to. The attempt to improve performance by updating the state in componentDidUpdate worked well, but the if a user navigated away from the Map page and navigated back, the Menu was blank. The workaround was to perform the filtering in the render method. This changes remove the work around and make use of the method getDerivedStateFromProps

kelvin-muchiri committed:
Update failing tests for utils/menuGroupHasVisibleLayers

kelvin-muchiri committed:
Remove debugger

kelvin-muchiri committed:
Do not adjust map height of hasNavBar prop is true in Map

Adjusting the height contributed to the legend sliding down after export and the legend content cut in the exported .png file. These changes make sure the map height is not adjusted, the legend bottom position is adjusted if hasNavBar prop is true.

kahummer committed:
Update state with latest props

kahummer committed:
Merge branch 'toggle-sectors-externally' of github.com:onaio/gisida-react into toggle-sectors-externally

kahummer committed:
Legend style tweaks

kahummer committed:
Harmonize Layers and connected layer params

kahummer committed:
Move HyperLinks to a component

kahummer committed:
Legend cleanup

kahummer committed:
Menu refactor and test updates

kahummer committed:
Hyperlink tests and component

kahummer committed:
Summary chart style tweaks

kahummer committed:
Build custom shape legend

kahummer committed:
Move layer type methods to utils

kahummer committed:
Update Layer test

kahummer committed:
Delete empty file

kahummer committed:
Remove unnecessary import

kahummer committed:
Cleanup

kahummer committed:
Merge branch 'toggle-sectors-externally' of github.com:onaio/gisida-react into toggle-sectors-externally
